browser-test/src/session_timeout.test.ts

```diff
@@ -0,0 +1,266 @@
+import {test, expect} from './support/civiform_fixtures'
+import {
+  AdminSettings,
+  enableFeatureFlag,
+  loginAsAdmin,
+  loginAsTestUser,
+  logout,
+  validateScreenshot,
+  validateToastMessage,
+} from './support'
+import {Page} from 'playwright'
+import {waitForHtmxReady} from './support/wait'
+
+type TimeoutData = {
+  inactivityWarning: number
+  inactivityTimeout: number
+  totalWarning: number
+  totalTimeout: number
+  currentTime: number
+}
+
+type TimeoutType = 'inactivityWarning' | 'totalWarning' | 'inactivityTimeout'
+
+/**
+ * Gets the adjusted timeout data that accounts for clock skew
+ * This matches what the SessionTimeoutHandler uses internally
+ */
+async function getAdjustedTimeoutData(page: Page): Promise<TimeoutData | null> {
```

Comment:
suggestion (blocking): Move this and other helper methods into a `support/session_timeout.ts` file. We generally try to keep test files just for tests and move helper methods into support files eg. https://github.com/civiform/civiform/blob/main/browser-test/src/support/admin_programs.ts